components/brave_today/browser/feed_building_unittest.cc

```diff
@@ -250,7 +272,51 @@ TEST(BraveNewsFeedBuilding, IncludesUserEnabledItems) {
           publisher_id_to_hide, "ESPN - Football", 14.525910905005045,
           "a minute ago")));
 
-  ASSERT_TRUE(ShouldDisplayFeedItem(feed_item, &publisher_list));
+  Channels channels;
+  ASSERT_TRUE(ShouldDisplayFeedItem(feed_item, &publisher_list, channels));
+}
+
+TEST_F(BraveNewsFeedBuildingTest, ChannelIsUsedWhenV2IsEnabled) {
+  // Enable the BraveNewsV2 Feature.
+  base::test::ScopedFeatureList features;
+  features.InitAndEnableFeature(brave_today::features::kBraveNewsV2Feature);
+
+  Publishers publisher_list;
+  PopulatePublishers(&publisher_list);
+  auto* publisher = publisher_list["111"].get();
+
+  Channels channels;
+  channels.insert(
+      {"Top News", brave_news::mojom::Channel::New("Top News", true)});
+  auto* channel = channels["Top News"].get();
+
+  auto feed_item = mojom::FeedItem::NewArticle(
+      mojom::Article::New(mojom::FeedItemMetadata::New(
+          "Technology", base::Time::Now(), "Title", "Description",
+          GURL("https://example.com/article"),
+          "7bb5d8b3e2eee9d317f0568dcb094850fdf2862b2ed6d583c62b2245ea507ab8",
+          mojom::Image::NewPaddedImageUrl(
+              GURL("https://example.com/article/image")),
+          publisher->publisher_id, "Source", 10, "a minute ago")));
+
+  // Publisher: NOT_MODIFIED, Channel: Subscribed, Should display.
+  ASSERT_TRUE(ShouldDisplayFeedItem(feed_item, &publisher_list, channels));
```

Comment:
Q) Did you pick `ASSERT_` series because below test doesn't have any meaning if it's failed?
If not `EXPECT_` would be better.

Reply:
Yeah, I generally prefer `EXPECT_` but this whole file was using `ASSERT_`. That said, `EXPECT_` does make more sense here, and I guess there's no reason to cargo cult :laughing: Changed.